Mage.Tests/src/test/java/org/mage/test/cards/abilities/other/MultipleAsThoughEffects.java

```diff
@@ -0,0 +1,68 @@
+package org.mage.test.cards.abilities.other;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * @author Alex-Vasile
+ */
+public class MultipleAsThoughEffects extends CardTestPlayerBase {
```

Comment:
Both of these tests still fail. For some reason the effects of all approving objects get applied.

Reply:
E.g. for the Bolas's Citadel test, even for the cast using Vivien the player still loses life equal to the mana cost.